src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -215,8 +217,19 @@ string GetTopicHref(TableOfContentsInputItem tocInputModel)
                 return default;
             }
 
-            (string resolvedTopicHref, List<string> monikers) ProcessTopicHref(string topicHref)
+            (string resolvedTopicHref, List<string> monikers) ProcessTopicItem(string uid, string topicHref)
             {
+                // process uid first
+                if (!string.IsNullOrEmpty(uid))
+                {
+                    var (resolvedTopicHref, monikers) = resolveXref.Invoke(filePath, uid);
```

Comment:
use `rootPath`?

Reply:
rootPath can make the error message not correct